homeassistant/components/homekit/accessories.py

```diff
@@ -105,8 +106,19 @@ def __init__(
         battery_found = self.hass.states.get(self.entity_id).attributes.get(
             ATTR_BATTERY_LEVEL
         )
-        if self.linked_battery_sensor:
-            battery_found = self.hass.states.get(self.linked_battery_sensor).state
```

Comment:
Can be simple fix if you check on this place if you get an None before you access to state. Your code work but look a bit complex for solve a simple problem like this

Reply:
I've simplified it down to a check, but I also want to display a warning for this scenario. Also, I needed `self.linked_battery_sensor = None` in order to make sure no errors are thrown during `run_handler`.

Reply:
```python
state = self.hass.states.get(self.linked_battery_sensor)
if state is not None:
    battery_found = state.state
else:
    …
```